COM-29 Add lab results form

lauravignoli opened the pull request.

lauravignoli committed:
COM-29 Add lab results form

tmvumbi2 commented:
@lauravignoli the error below is displayed when loading the form; I can see the concept referred to in the list of concepts added, not sure what's causing the issue

![image](https://user-images.githubusercontent.com/22587098/56565618-2112ae80-65b1-11e9-81d7-5d8c211131f6.png)

lauravignoli commented:
@tmvumbi2 Yes, I added a comment for that issue here https://github.com/jembi/cameroon-bahmni-config/pull/37, haven't managed to find why it's happening

tmvumbi2 commented:
@lauravignoli the date field isn't displayed when a result value is captured for `Epithelial cells per field`

![image](https://user-images.githubusercontent.com/22587098/56565960-da718400-65b1-11e9-8b6e-5e697809914c.png)

tmvumbi2 commented:
@lauravignoli the date field isn't displayed for `Trichomonas vaginalis` as well 

![image](https://user-images.githubusercontent.com/22587098/56566458-fde8fe80-65b2-11e9-99cd-dabc44971411.png)

tmvumbi2 commented:
@lauravignoli The date field isn't displayed for `Total cholesterol` and `Triglycerides` as well:

![image](https://user-images.githubusercontent.com/22587098/56567478-50c3b580-65b5-11e9-89f4-d46cb22f0e6b.png)

tmvumbi2 commented:
@lauravignoli same feedback for `HDL`

![image](https://user-images.githubusercontent.com/22587098/56567712-ca5ba380-65b5-11e9-963f-7d3539cc939a.png)

tmvumbi2 commented:
@lauravignoli the field `Platelets` should accept any text value according to the CSV (it's numeric with a unit and range on the form)

![image](https://user-images.githubusercontent.com/22587098/56568106-a77dbf00-65b6-11e9-94c0-f8a453f1d568.png)

tmvumbi2 commented:
@lauravignoli `HIV Test` should be a section title according to the CSV, it's a Yes/No question on the form:

![image](https://user-images.githubusercontent.com/22587098/56568462-5ae6b380-65b7-11e9-86c3-ec70c4090b8d.png)

tmvumbi2 commented on configuration/concepts/lab_results_form.csv:
```diff
@@ -0,0 +1,334 @@
+baf5d5ac-47a0-4733-b58c-dfc8ef64530f,,,Skin snip,Skin snip,Skin snip,Skin snip,LabTest,Text,,
+4d4d0e97-af80-4e58-a63d-6739d133b4ec,,,Skin snip Date de résultat,Skin snip Test Date,Date de résultat,Test Date,Misc,Date,,
+7479640d-b96d-442c-be05-542bcd009a65,Crachants;Crachants Test Date;CSF parasitology;CSF parasitology Test Date;Coprology;Coprology Test Date;Rapid Diagnosis Malaria Test;Rapid Diagnosis Malaria Test Test Date;Malaria Test (GE);Malaria Test (GE) Test Date;Recherche des Micro Filaires;Recherche des Micro Filaires Test Date;Skin snip;Skin snip Test Date,,Département de Parasitologie,Parasitology Dep ...
+3b5b8d72-2f86-48fc-9b81-0be908ed392c,Bacteriology Dept;Haematology Dept;Biochemistry Dept;Parasitology Dept,,Résultats de laboratoire - ajoutés manuellement,Lab results - add manually,Résultats de laboratoire - ajoutés manuellement,Lab results - add manually,Misc,N/A,,
```
@lauravignoli Please use full caps for this concept's name

lauravignoli replied:
Fixed

tmvumbi2 reviewed (changes requested):
Thanks @lauravignoli, the PR looks great overall. I left a few feedbacks

lauravignoli committed:
COM-29 Capitalize lab entry form name

lauravignoli committed:
COM-29 Fix import issues

Renamed HIV test to HIV Tests as it was conflicting with another concept.

Assigned a new uuid to CSF as it was duplicated with CSF parasitology.

lauravignoli commented:
> @lauravignoli the error below is displayed when loading the form; I can see the concept referred to in the list of concepts added, not sure what's causing the issue
> 
> ![image](https://user-images.githubusercontent.com/22587098/56565618-2112ae80-65b1-11e9-81d7-5d8c211131f6.png)

Fixed

lauravignoli commented:
> @lauravignoli `HIV Test` should be a section title according to the CSV, it's a Yes/No question on the form:
> 
> ![image](https://user-images.githubusercontent.com/22587098/56568462-5ae6b380-65b7-11e9-86c3-ec70c4090b8d.png)

Fixed

lauravignoli commented:
@tmvumbi2 Fixed everything except the Total cholesterol and Tryglicerids, these are duplicated concepts, I'm busy finding a way to differenciate them in the form.

I used the existing concept for Platelets instead of creating a Text new one. Will check with Lisa.

lauravignoli committed:
COM-29 Differentiated concepts

We don't have a way to apply logic for concepts that are used more than once in the form yet. Total cholesterol and Triglycerides were used twice, and had to be differentiated for the logic to work.

lauravignoli committed:
COM-29 Fixed CSF concept

Uuid was wrong.
